File: app/src/main/java/com/jakubokrasa/uprightchallenge/data/PostureStatDao.java
package com.jakubokrasa.uprightchallenge.data;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.List;


@Dao
public interface PostureStatDao {
    @Insert
    void insert(PostureStat postureStat);

    @Query("SELECT * from PostureStat ORDER BY statId ASC")
    List<PostureStat> getAllStats();

    @Query("DELETE FROM PostureStat")
    void deleteAll();


}

File: app/src/main/java/com/jakubokrasa/uprightchallenge/data/PostureStatDao.kt
package com.jakubokrasa.uprightchallenge.data

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query

@Dao
interface PostureStatDao {
    @Insert
    suspend fun insertPostureStat(postureStat: PostureStat?)

    @Query("SELECT * from PostureStat ORDER BY statId ASC")
    fun getAllStats(): LiveData<List<PostureStat>>

    @Query("DELETE FROM PostureStat")
    suspend fun deleteAllStats()
}